src/clients/HubPoolClient.ts

```diff
@@ -918,21 +918,29 @@ export class HubPoolClient extends BaseAbstractClient {
     if (eventsToQuery.includes("SetPoolRebalanceRoute")) {
       for (const event of events["SetPoolRebalanceRoute"]) {
         const args = spreadEventWithBlockNumber(event) as SetPoolRebalanceRoot;
-        assign(this.l1TokensToDestinationTokens, [args.l1Token, args.destinationChainId], args.destinationToken);
-        assign(
-          this.l1TokensToDestinationTokensWithBlock,
-          [args.l1Token, args.destinationChainId],
-          [
-            {
-              l1Token: args.l1Token,
-              l2Token: args.destinationToken,
-              blockNumber: args.blockNumber,
-              transactionIndex: args.transactionIndex,
-              logIndex: args.logIndex,
-              transactionHash: args.transactionHash,
-            },
-          ]
-        );
+        // If the destination token is set to the zero address in an event, then this means Across should no longer
+        // rebalance to this chain.
+        if (args.destinationToken !== ZERO_ADDRESS) {
+          assign(this.l1TokensToDestinationTokens, [args.l1Token, args.destinationChainId], args.destinationToken);
+          assign(
+            this.l1TokensToDestinationTokensWithBlock,
+            [args.l1Token, args.destinationChainId],
+            [
+              {
+                l1Token: args.l1Token,
+                l2Token: args.destinationToken,
+                blockNumber: args.blockNumber,
+                transactionIndex: args.transactionIndex,
+                logIndex: args.logIndex,
+                transactionHash: args.transactionHash,
+              },
+            ]
+          );
+        } else {
+          // Clear out the mapping for the L1 token entirely.
+          delete this.l1TokensToDestinationTokens[args.l1Token][args.destinationChainId];
```

Comment:
This might lead to some where artifacts where `this.l1TokensToDestinationTokens[args.l1Token]` is an empty dict because all of its destination chains are deleted--not sure if there are any problems downstream though

Reply:
It probably doesn't hurt to delete these too. I'm sure we iterate over these values, https://github.com/across-protocol/sdk/pull/953/commits/c864ee6a0350a22f99f4a785f1e34913b01220aa